# HG changeset patch
# User "Cole Robinson <crobinso@redhat.com>"
# Date 1223313666 14400
# Node ID 270e1697b81ac143df54e111e2e1bed5006ad49c
# Parent  9f5d5b6940c5a266f7b3a087f41face4713feec2
populate hostinfo earlier in tick function so it isn't accessed uninitialized.

diff -r 9f5d5b6940c5 -r 270e1697b81a src/virtManager/connection.py
--- a/src/virtManager/connection.py	Mon Oct 06 12:56:11 2008 -0400
+++ b/src/virtManager/connection.py	Mon Oct 06 13:21:06 2008 -0400
@@ -869,6 +869,8 @@
         if self.state != self.STATE_ACTIVE:
             return
 
+        self.hostinfo = self.vmm.getInfo()
+
         # Poll for new virtual network objects
         (startNets, stopNets, newNets,
          oldNets, self.nets) = self._update_nets()
@@ -911,10 +913,6 @@
 
         # Finally, we sample each domain
         now = time()
-        try:
-            self.hostinfo = self.vmm.getInfo()
-        except:
-            logging.warn("Unable to get host information")
 
         updateVMs = self.vms
         if noStatsUpdate:
